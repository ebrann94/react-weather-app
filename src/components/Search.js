import React from 'react';

const Search = ({ handleLocation, apiError }) => (
    <div className="location-input">
        {apiError && <p className="api-error">Location not found, please enter another!</p>}
        <form onSubmit={handleLocation} className="location-form">
            <input type="text" name="location" placeholder="Search" className="location-input__text" />
            <input type="image" name="submit" src='./images/search.svg' className="search-icon" borders="0"  />
        </form>
    </div>
);

export default Search;