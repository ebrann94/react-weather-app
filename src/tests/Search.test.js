import React from 'react';
import { shallow, mount } from 'enzyme';
import Search from '../components/Search';
import { isFlowBaseAnnotation } from '@babel/types';

describe('Search component tests', () => {
    it('renders without crashing', () => {
        const props = {
            handleLocation: jest.fn(),
            apiError: false
        }
        const wrapper = shallow(<Search {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('updates state on input change', () => {
        const props = {
            handleLocation: jest.fn(),
            apiError: false
        }
        const event = {
            target: {
                value: 'London'
            }
        }
        const wrapper = mount(<Search {...props} />);
        expect(wrapper.find('input.location-input__text').props().value).toBe('');
        wrapper.find('input.location-input__text').simulate('change', event);
        expect(wrapper.find('input.location-input__text').props().value).toBe('London');
    });

    it('calls handleLocation on submit with location input', () => {
        const props = {
            handleLocation: jest.fn(),
            apiError: false
        }
        const changeEvent = {
            target: {
                value: 'London'
            }
        }
        const submitEvent = {
            preventDefault() {},
            target: {
                location: {
                    blur() {}
                }
            }
        }
        const wrapper = mount(<Search {...props}/>);
        wrapper.find('input.location-input__text').simulate('change', changeEvent);
        wrapper.find('form.location-form').simulate('submit', submitEvent);
        expect(props.handleLocation).toHaveBeenCalledWith('London');
    });

    it('renders error message when error', () => {
        const props = {
            handleLocation() {},
            apiError: true
        }
        const wrapper = shallow(<Search {...props} />);
        expect(wrapper.find('p.api-error').exists()).toBe(true);
    });
});