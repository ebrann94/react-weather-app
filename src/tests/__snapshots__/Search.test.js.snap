// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component tests renders without crashing 1`] = `ShallowWrapper {}`;
